### ISSUE TITLE
OpenJDK LockStep Mauve plus AIX RacingCollections SyncSubMutexes Map/Defaults crash vmState=0x00000000

### ISSUE THREAD

**Issue by pshipton** [eclipse-openj9/openj9#14465]
https://openj9-jenkins.osuosl.org/job/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/2 - [cent7-ppcle-2](https://openj9-jenkins.osuosl.org/computer/cent7-ppcle-2)
https://openj9-jenkins.osuosl.org/job/Test_openjdk18_j9_sanity.openjdk_ppc64_aix_Nightly/2 - [p8-java1-ibm06](https://openj9-jenkins.osuosl.org/computer/p8-java1-ibm06)

java/util/NavigableMap/LockStep.java - both plinux and AIX

AIX only:
java/util/Collections/RacingCollections.java
java/util/Collections/SyncSubMutexes.java
java/util/Map/Defaults.java

https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Test/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/2/openjdk_test_output.tar.gz

https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Test/Test_openjdk18_j9_sanity.openjdk_ppc64_aix_Nightly/2/openjdk_test_output.tar.gz

ppc64le jdk_util_0
```
19:42:47  rerun:
19:42:47  cd /home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/TKG/output_16443625782194/jdk_util_0/work/scratch/1 && \
19:42:47  DISPLAY=:0 \
19:42:47  HOME=/home/jenkins \
19:42:47  LANG=en_US.UTF-8 \
19:42:47  PATH=/bin:/usr/bin:/usr/sbin \
19:42:47      /home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/openjdkbinary/j2sdk-image/bin/java \
19:42:47          -Dtest.vm.opts='-ea -esa -Xmx1540m -Xdump:system:none -Xdump:heap:none -Xdump:system:events=gpf+abort+traceassert+corruptcache -XX:+UseCompressedOops' \
19:42:47          -Dtest.tool.vm.opts='-J-ea -J-esa -J-Xmx1540m -J-Xdump:system:none -J-Xdump:heap:none -J-Xdump:system:events=gpf+abort+traceassert+corruptcache -J-XX:+UseCompressedOops' \
19:42:47          -Dtest.compiler.opts= \
19:42:47          -Dtest.java.opts= \
19:42:47          -Dtest.jdk=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/openjdkbinary/j2sdk-image \
19:42:47          -Dcompile.jdk=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/openjdkbinary/j2sdk-image \
19:42:47          -Dtest.timeout.factor=8.0 \
19:42:47          -Dtest.nativepath=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/openjdkbinary/openjdk-test-image/jdk/jtreg/native \
19:42:47          -Dtest.root=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/openjdk/openjdk-jdk/test/jdk \
19:42:47          -Dtest.name=java/util/NavigableMap/LockStep.java \
19:42:47          -Dtest.file=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/openjdk/openjdk-jdk/test/jdk/java/util/NavigableMap/LockStep.java \
19:42:47          -Dtest.src=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/openjdk/openjdk-jdk/test/jdk/java/util/NavigableMap \
19:42:47          -Dtest.src.path=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/openjdk/openjdk-jdk/test/jdk/java/util/NavigableMap \
19:42:47          -Dtest.classes=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/TKG/output_16443625782194/jdk_util_0/work/classes/2/java/util/NavigableMap/LockStep.d \
19:42:47          -Dtest.class.path=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/TKG/output_16443625782194/jdk_util_0/work/classes/2/java/util/NavigableMap/LockStep.d \
19:42:47          -Dtest.class.path.prefix=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/TKG/output_16443625782194/jdk_util_0/work/classes/2/java/util/NavigableMap/LockStep.d:/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/openjdk/openjdk-jdk/test/jdk/java/util/NavigableMap \
19:42:47          -classpath /home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/TKG/output_16443625782194/jdk_util_0/work/classes/2/java/util/NavigableMap/LockStep.d:/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/aqa-tests/openjdk/openjdk-jdk/test/jdk/java/util/NavigableMap:/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/jvmtest/openjdk/jtreg/lib/javatest.jar:/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/jvmtest/openjdk/jtreg/lib/jtreg.jar \
19:42:47          LockStep
19:42:47  stderr:
19:42:47  Unhandled exception
19:42:47  Type=Segmentation error vmState=0x00000000
19:42:47  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
19:42:47  Handler1=00003FFFA6E456B0 Handler2=00003FFFA6D7C3A0
19:42:47  R0=00000000000CD600 R1=00003FFF7EFEB750 R2=00000000FFC30558 R3=00000000FFC30558
19:42:47  R4=000000000012BB00 R5=00000000004C8E00 R6=00000000FFC2EC48 R7=00003FFF7DA60038
19:42:47  R8=0000000000534500 R9=00000000000CE7E0 R10=00000000004F1A00 R11=0000000000000000
19:42:47  R12=F9EBFFFFC0005B0B R13=00003FFF7EFF6900 R14=0000000000534500 R15=00000000004C8E00
19:42:47  R16=00003FFF7DA60038 R17=00000000FFC2EC48 R18=000000009FC73948 R19=00000000FFA2C3C8
19:42:47  R20=00000000FFC2ECC0 R21=000000009FEBE958 R22=00000000FFA2C4F8 R23=00000000FFC2EC48
19:42:47  R24=00000000FFC2D008 R25=0000000000000007 R26=00000000FFB0C4B8 R27=00000000FFA2C3C8
19:42:47  R28=0000000000541500 R29=0000000000448100 R30=00000000FFE9D7A8 R31=00000000004F1A00
19:42:47  NIP=00003FFFA60B75C8 MSR=800000010280F033 ORIG_GPR3=00000000000081C8 CTR=00003FFFA60B7578
19:42:47  LINK=00003FFF7F85C798 XER=0000000020000000 CCR=0000000024004424 SOFTE=0000000000000001
19:42:47  TRAP=0000000000000300 DAR=F9EBFFFFC04F750B dsisr=0000000040000000 RESULT=0000000000000000
19:42:47  FPR0 0000000000534500 (f: 5457152.000000, d: 2.696191e-317)
19:42:47  FPR1 4054235360000000 (f: 1610612736.000000, d: 8.055196e+01)
19:42:47  FPR2 4008000000000000 (f: 0.000000, d: 3.000000e+00)
19:42:47  FPR3 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR4 3fe3333340000000 (f: 1073741824.000000, d: 6.000000e-01)
19:42:47  FPR5 bfc4ebe6a0000000 (f: 2684354560.000000, d: -1.634491e-01)
19:42:47  FPR6 3fe5555560000000 (f: 1610612736.000000, d: 6.666667e-01)
19:42:47  FPR7 3f28eea840000000 (f: 1073741824.000000, d: 1.902180e-04)
19:42:47  FPR8 003600330034003d (f: 3407933.000000, d: 1.223834e-307)
19:42:47  FPR9 003000350020002c (f: 2097196.000000, d: 8.900745e-308)
19:42:47  FPR10 00300035003d0031 (f: 3997745.000000, d: 8.900745e-308)
19:42:47  FPR11 380f400c80000000 (f: 2147483648.000000, d: 1.147951e-38)
19:42:47  FPR12 410bcf2800000000 (f: 0.000000, d: 2.278130e+05)
19:42:47  FPR13 3fdb9b2820000000 (f: 536870912.000000, d: 4.313450e-01)
19:42:47  FPR14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR16 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR17 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR19 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  FPR31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:42:47  Module=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/openjdkbinary/j2sdk-image/lib/default/libj9jit29.so
19:42:47  Module_base_address=00003FFFA5AF0000
19:42:47  Target=2_90_20220208_9 (Linux 3.10.0-1127.18.2.el7.ppc64le)
19:42:47  CPU=ppc64le (8 logical CPUs) (0x1de2b0000 RAM)
19:42:47  ----------- Stack Backtrace -----------
19:42:47  (0x00003FFFA6D76A84 [libj9prt29.so+0x36a84])
19:42:47  (0x00003FFFA6D7D7A8 [libj9prt29.so+0x3d7a8])
19:42:47  (0x00003FFFA6D76B48 [libj9prt29.so+0x36b48])
19:42:47  (0x00003FFFA6D76CB0 [libj9prt29.so+0x36cb0])
19:42:47  (0x00003FFFA6D76784 [libj9prt29.so+0x36784])
19:42:47  (0x00003FFFA6D7D7A8 [libj9prt29.so+0x3d7a8])
19:42:47  (0x00003FFFA6D76878 [libj9prt29.so+0x36878])
19:42:47  (0x00003FFFA6E45530 [libj9vm29.so+0x45530])
19:42:47  (0x00003FFFA6D7D7A8 [libj9prt29.so+0x3d7a8])
19:42:47  (0x00003FFFA6E45864 [libj9vm29.so+0x45864])
19:42:47  (0x00003FFFA6D7C600 [libj9prt29.so+0x3c600])
19:42:47  __kernel_sigtramp_rt64+0x0 (0x00003FFFA7780478)
19:42:47  (0x00003FFFA6F6FA08 [libj9vm29.so+0x16fa08])
19:42:47  (0x00003FFFA6E19BD0 [libj9vm29.so+0x19bd0])
19:42:47  (0x00003FFFA6EB79F0 [libj9vm29.so+0xb79f0])
19:42:47  (0x00003FFFA6D7D7A8 [libj9prt29.so+0x3d7a8])
19:42:47  (0x00003FFFA6EB2B84 [libj9vm29.so+0xb2b84])
19:42:47  (0x00003FFFA6D113E8 [libj9thr29.so+0x113e8])
19:42:47  (0x00003FFFA76B8CD4 [libpthread.so.0+0x8cd4])
19:42:47  clone+0xe4 (0x00003FFFA75B7F14 [libc.so.6+0x127f14])
19:42:47  ---------------------------------------
```

ppc64le jdk_util_1
```
20:07:49  command: main -XX:+IgnoreUnrecognizedVMOptions -XX:+EliminateAutoBox -XX:AutoBoxCacheMax=20000 -Dthorough=true LockStep
20:07:49  reason: User specified action: run main/othervm -XX:+IgnoreUnrecognizedVMOptions -XX:+EliminateAutoBox -XX:AutoBoxCacheMax=20000 -Dthorough=true LockStep 
20:07:49  Mode: othervm [/othervm specified]
20:07:49  elapsed time (seconds): 16.946
20:07:49  configuration:
20:07:49  STDOUT:
20:07:49  LockStep: Trial random seed: 47705875224597587
20:07:49  STDERR:
20:07:49  Unhandled exception
20:07:49  Type=Segmentation error vmState=0x00000000
20:07:49  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
20:07:49  Handler1=00003FFF9C1C56B0 Handler2=00003FFF9C0FC3A0
20:07:49  R0=00003FFF981D0F00 R1=00003FFF0DD9B750 R2=00003FFF8FE8BC38 R3=00003FFF8FE8BC38
20:07:49  R4=00003FFF9823B700 R5=00003FFF983A2C00 R6=00003FFF2FB70000 R7=0000000000600000
20:07:49  R8=0000000000000000 R9=00003FFF981D20E0 R10=00003FFEA0142000 R11=0000000000000000
20:07:49  R12=30ABFFFFC000705F R13=00003FFF0DDA6900 R14=00003FFEA000C570 R15=00003FFF983A2C00
20:07:49  R16=00003FFF0E990038 R17=00003FFF8FE41780 R18=00003FFF2FD0C298 R19=00003FFF8FE3A358
20:07:49  R20=00003FFF8FE418F8 R21=00003FFF8FE3A358 R22=00003FFEA000C6A8 R23=00003FFEA000C6A8
20:07:49  R24=00003FFF8FE86FA8 R25=00003FFF8FE86948 R26=00003FFF8FE85558 R27=00003FFF8FE7FC30
20:07:49  R28=00003FFEA01A8400 R29=00003FFEA00AFB00 R30=00003FFF8FE38F10 R31=00003FFEA0142000
20:07:49  NIP=00003FFF90B775C8 MSR=800000010280F033 ORIG_GPR3=00000000000081C8 CTR=00003FFF90B77578
20:07:49  LINK=00003FFF15377218 XER=0000000020000000 CCR=0000000024004422 SOFTE=0000000000000001
20:07:49  TRAP=0000000000000300 DAR=30AC3FFE6014905F dsisr=0000000040000000 RESULT=0000000000000000
20:07:49  FPR0 00003ffea000c620 (f: 2684405248.000000, d: 3.476386e-310)
20:07:49  FPR1 4052239180000000 (f: 2147483648.000000, d: 7.255576e+01)
20:07:49  FPR2 3fbf848580000000 (f: 2147483648.000000, d: 1.231159e-01)
20:07:49  FPR3 3fee666660000000 (f: 1610612736.000000, d: 9.500000e-01)
20:07:49  FPR4 400bd52000000000 (f: 0.000000, d: 3.479065e+00)
20:07:49  FPR5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR6 3fe5555560000000 (f: 1610612736.000000, d: 6.666667e-01)
20:07:49  FPR7 3f268ec900000000 (f: 0.000000, d: 1.721020e-04)
20:07:49  FPR8 00003fff8fe899e8 (f: 2414385664.000000, d: 3.476585e-310)
20:07:49  FPR9 00003fff8fe89a20 (f: 2414385664.000000, d: 3.476585e-310)
20:07:49  FPR10 00003fff8fe89850 (f: 2414385152.000000, d: 3.476585e-310)
20:07:49  FPR11 3ff0000000000000 (f: 0.000000, d: 1.000000e+00)
20:07:49  FPR12 40e9d70000000000 (f: 0.000000, d: 5.292000e+04)
20:07:49  FPR13 4020800000000000 (f: 0.000000, d: 8.250000e+00)
20:07:49  FPR14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR16 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR17 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR19 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  FPR31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
20:07:49  Module=/home/jenkins/workspace/Test_openjdk18_j9_sanity.openjdk_ppc64le_linux_Nightly/openjdkbinary/j2sdk-image/lib/default/libj9jit29.so
20:07:49  Module_base_address=00003FFF905B0000
20:07:49  Target=2_90_20220208_9 (Linux 3.10.0-1127.18.2.el7.ppc64le)
20:07:49  CPU=ppc64le (8 logical CPUs) (0x1de2b0000 RAM)
20:07:49  ----------- Stack Backtrace -----------
20:07:49  (0x00003FFF9C0F6A84 [libj9prt29.so+0x36a84])
20:07:49  (0x00003FFF9C0FD7A8 [libj9prt29.so+0x3d7a8])
20:07:49  (0x00003FFF9C0F6B48 [libj9prt29.so+0x36b48])
20:07:49  (0x00003FFF9C0F6CB0 [libj9prt29.so+0x36cb0])
20:07:49  (0x00003FFF9C0F6784 [libj9prt29.so+0x36784])
20:07:49  (0x00003FFF9C0FD7A8 [libj9prt29.so+0x3d7a8])
20:07:49  (0x00003FFF9C0F6878 [libj9prt29.so+0x36878])
20:07:49  (0x00003FFF9C1C5530 [libj9vm29.so+0x45530])
20:07:49  (0x00003FFF9C0FD7A8 [libj9prt29.so+0x3d7a8])
20:07:49  (0x00003FFF9C1C5864 [libj9vm29.so+0x45864])
20:07:49  (0x00003FFF9C0FC600 [libj9prt29.so+0x3c600])
20:07:49  __kernel_sigtramp_rt64+0x0 (0x00003FFF9CB00478)
20:07:49  (0x00003FFF9119B860 [libj9jit29.so+0xbeb860])
20:07:49  (0x00003FFF9C199BD0 [libj9vm29.so+0x19bd0])
20:07:49  (0x00003FFF9C2379F0 [libj9vm29.so+0xb79f0])
20:07:49  (0x00003FFF9C0FD7A8 [libj9prt29.so+0x3d7a8])
20:07:49  (0x00003FFF9C232B84 [libj9vm29.so+0xb2b84])
20:07:49  (0x00003FFF9C0913E8 [libj9thr29.so+0x113e8])
20:07:49  (0x00003FFF9CA38CD4 [libpthread.so.0+0x8cd4])
20:07:49  clone+0xe4 (0x00003FFF9C937F14 [libc.so.6+0x127f14])
20:07:49  ---------------------------------------
```

**Comment by pshipton** [comment]
@0xdaryl still some crashes on jdk18 sanity.openjdk on P.

**Comment by 0xdaryl** [comment]
@zl-wang : could you have someone on Power triage these failures please?

**Comment by zl-wang** [comment]
@ymanton help is needed here

**Comment by pshipton** [comment]
On AIX in the same build above, there are also crashes in the following tests:
java/util/Collections/SyncSubMutexes.java
java/util/Map/Defaults.java

**Comment by pshipton** [comment]
sanity.system Mauve testing is also failing on plinux and AIX, guessing it's the same issue.
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-JDK18/14/

**Comment by ymanton** [comment]
This is the same problem described by https://github.com/eclipse-openj9/openj9/issues/14228 which itself is likely a dup of https://github.com/eclipse-openj9/openj9/issues/14230 for which a fix was recently merged.

The crash I looked at is during an interface call to `java/util/Collection.toString()Ljava/lang/String;` from an AOT-compiled method `java/util/Collections$SynchronizedCollection.toString()Ljava/lang/String;`. The thread crashes in `_interfaceSlotsUnavailable` because we have an `iTable` index that's out of bounds.

**Comment by pshipton** [comment]
@ymanton pls create a PR to double deliver the fix to the v0.31.0-release branch.

**Comment by pshipton** [comment]
If you are talking about https://github.com/eclipse-openj9/openj9/pull/14430, these problems persist after that fix.

**Comment by ymanton** [comment]
#14230 is caused by the optimizer transforming an interface call (that needs to be resolved at runtime because we're in AOT mode) to a virtual call, thereby creating a mismatch between the constant pool entry (which still describes an interface call) and the generated code, which causes problems at resolve time.

In this case we jave an interface call and we've kept it intact (possibly due to #14430 kicking in) and generated the expected code but something is still going wrong. I'm going to try tracking down the source of the bad iTable index.

**Comment by 0xdaryl** [comment]
Hi @ymanton...  Could you post a short update on any progress on this blocking issue please?

**Comment by ymanton** [comment]
I followed the trail of the iTable index back into the VM and discovered that the low bits of of value distinguish between whether the value is an iTable or vTable index. It looks like we did check the low bits, but the check (written in assembly) is subtly incorrect. The check is:

```
        ! lastITable is a match
        laddr   r12, 4*ALen(r11)                                ! Load the itable offset from the snippet
        andis.  r11,r12,J9TR_J9_ITABLE_OFFSET_TAG_BITS          ! Call the helper if the itable offset is tagged
        bne     .L.callHelper
```

It uses the `andis.` instruction, which shifts the mask `<<16` before applying it, thereby no longer testing the intended bits. It should be `andi.` instead.

**Comment by pshipton** [comment]
This was closed by https://github.com/eclipse-openj9/openj9/pull/14527 but PR testing in that PR shows the failures in this issue still occur. Reopening.

**Comment by pshipton** [comment]
The only test mentioned in this issue not seen in the PR testing for https://github.com/eclipse-openj9/openj9/pull/14527 is java/util/Collections/SyncSubMutexes.java on AIX so perhaps that one is fixed now.

**Comment by ymanton** [comment]
Looks like there's another bug near the one I fixed in #14527. When we do the iTable low bits check we clobber `r11` (replacing the pointer to the snippet data with the low bits of the iTable offset) but the snippet pointer is still required on the path that handles the case where the low bits are set as it is eventually dereferenced by `old_fast_jitLookupInterfaceMethod`.

The changes to this code were done 4 years ago in https://github.com/eclipse-openj9/openj9/commit/0bda1a32b865095b704686bd1a04b59a45bed058 for private interface methods, but it seems like this particular path hasn't ever been executed by anyone until now. To get to this path you have to 1) miss the 2 interface cache slots at the call site but 2) hit the last iTable cache slot in the receiver class and 3) have the two low bits of the iTable offset set. 1 and 2 are common occurrences, and I assume 3 by itself is, but the combination hasn't been until now.
